src/commands/deploy/pipeline.ts

```diff
@@ -43,4 +43,18 @@ export default class DeployPipeline extends PromoteCommand<typeof SfCommand> {
       ? pipelineStage.sf_devops__Pipeline_Stages__r.records[0].Id
       : 'Approved';
   }
+
+  /**
+   * Default function for catching commands errors.
+   *
+   * @param error
+   * @returns
+   */
+  protected catch(error: Error | SfError): Promise<SfCommand.Error> {
```

Comment:
Since this behivor is common to all implementations couldn't we moved this logic to the abstract class and then delegate the generation of the error message in the concrete classes?